crypto/threads_pthread.c

```diff
@@ -188,7 +188,7 @@ int CRYPTO_THREAD_compare_id(CRYPTO_THREAD_ID a, CRYPTO_THREAD_ID b)
 
 int CRYPTO_atomic_add(int *val, int amount, int *ret, CRYPTO_RWLOCK *lock)
 {
-# if defined(__GNUC__) && defined(__ATOMIC_ACQ_REL)
+# if defined(__GNUC__) && defined(__ATOMIC_ACQ_REL) && !(defined(__apple_build_version__) && 6000000 > __apple_build_version__)
```

Comment:
there's no information here about why this is being done, which could be a problem for future maintainers of this file

would it be better to do something like

```
#if defined(__apple_build_version__) && __apple_build_version__ < 6000000
// OS/X 10.7 and 10.8 had a weird version of clang which has __ATOMIC_ACQUIRE and
// __ATOMIC_ACQ_REL but which expects only one parameter for __atomic_add_fetch()
// rather than two
#define BROKEN_CLANG_ATOMICS
#endif
```

at the top, and

```
# if defined(__GNUC__) && defined(__ATOMIC_ACQ_REL) && !defined(BROKEN_CLANG_ATOMICS)
```

here?

Reply:
@tom-cosgrove-arm very good idea, thanks!